src/index.ts

```diff
@@ -956,7 +955,7 @@ export default class Purchases {
    *
    * @warning This function should only be called if you're not calling makePurchase.
    */
-  public static syncPurchases() {
+  public static syncPurchases(): void {
```

Comment:
@aboedo @vegaro I noticed your sidebar discussion related to whether `syncPurchases()` should call `getPurchaserInfo()` as well. So I have a little follow-up to this one here, as it comes up in our current SDK Implementation:

If `syncPurchases()` does not return an (async) response on whether synching succeeded or failed, it is impossible for an SDK consumer to respond accordingly (even to know whether the async action of syncPurchases() has finished).

So there is no way for the App to know, whether synched current PurchaserInfo is available?

If synching on top of that does not "automatically" trigger a firing of a purchases listener attached with `addPurchaserInfoUpdateListener(...)`, then there is no good way for the App to respond to the sync attempt completion.

So, this makes me wonder:
* Should syncing return `Promise<PurchaserInfo>` and reject if it fails.
* Should it also always ensure that the purchases listener fires? As this would be consistent with the behaviour of e.g. `purchasePackage(...)`?

Reply:
Thanks for weighing in here as well. 

Syncing should always trigger an update to the listener if any changes to purchaserInfo are made, so that part would still work in either case.
I agree that syncing should return `Promise<PurchaserInfo>`, we have that in the backlog. That part however requires an update to `purchases-android` and `purchases-hybrid-common`, which would take a bit longer, so that's why I proposed having the correct return type as a first step. 

@vegaro's suggestion would serve as a temporary workaround to ensure that the method could always return a `Promise<PurchaserInfo>`, with the disadvantage being that we'd be making an extra backend call. 

Thinking about this again, perhaps it's better to start with @vegaro's workaround and then improve the underlying implementation, so that we're not updating the interface again afterwards.

Reply:
Thanks for humoring me and being so responsive, really appreciated 🙇

Reply:
Thinking about this more, @vegaro's suggestion wouldn't really work, since we don't have a clean way to know when `syncPurchases` is done in the first place. The correct solution to this is to make it asynchronous in `purchases-android` first, then make this a promise (or add a new method and deprecate this one). 
For now, though, we're going to have this just return right away, for consistency.